vm/src/memory/constant_pool.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include "errors.h"
+#include "memory/strings.h"
+#include <memory>
+#include <vector>
+
+class ByteReader;
+
+
+/**
+ * defines the type for an index inside the constant pool.
+ * */
+typedef uint32_t constant_index;
+
+/**
+ * Kind of constant that can be contained in the constant pool
+ * */
+enum PoolConstantType {
+    /// an utf-8 string
+    C_STR,
+};
```

Comment:
```suggestion
};

std::ostream &operator<<(std::ostream &os, const PoolConstantType &type);
```

Sample implementation:
```cpp
std::ostream &operator<<(std::ostream &os, const PoolConstantType &type) {
    switch (type) {
    case PoolConstantType::C_STR:
        return os << "C_STR";
    default:
        return os << "UNKNOWN";
    }
}
```